e2e/pages/filing-app/formAlerts.spec.ts

```diff
@@ -0,0 +1,259 @@
+          '../../test-data/sample-sblar-files/errors-page-1-syntax-few.csv',
+        ),
+      );
+      await expect(page.getByText('File upload in progress')).toBeVisible();
+      await expect(page.getByText('File upload successful')).toBeVisible({
+        timeout: 10_000,
+      });
+      await expect(page.getByText('Validation checks in progress')).toBeVisible(
+        {
+          timeout: 10_000,
+        },
+      );
+      await expect(
+        page.getByText('Errors were found in your file'),
+      ).toBeVisible({ timeout: 60_000 });
+    });
+
+    // Continue to next page
+    await test.step('Navigate: Resolve errors (syntax)', async () => {
+      await page.waitForSelector('#nav-next');
+      await page.waitForTimeout(500);
+      await page.getByRole('button', { name: 'Continue to next step' }).click();
+    });
+  });
+
+  // Resolve errors (syntax) page
+  await test.step('Resolve errors (syntax) page', async () => {
+    await expect(page.locator('h1')).toContainText('Resolve errors (syntax)');
+    await expect(page.locator('#error-header-alert')).toContainText(
+      'Your register contains syntax errorsThere may be an issue with the data type or format of one or more values in your file. Make sure your register meets the requirements detailed in the filing instructions guide (section 4, "Data validation"), make the corrections, and upload a new file.',
+    );
+    await page
+      .getByRole('button', { name: 'Continue' })
+      .click({ timeout: 500 });
+
+    await expect(page.locator('#error-footer-alert')).toContainText(
+      'You must resolve syntax errors to continue.',
+    );
+
+    await page.getByRole('link', { name: 'Upload a new file' }).click();
+  });
+
+  // Upload file page
+  await test.step('Upload file page', async () => {
+    await expect(page.locator('h1')).toContainText('Upload file');
+
+    // Upload file with syntax errors
+    await test.step('Upload file: (errors-page-2-logic-few.csv)', async () => {
+      await expect(page.locator('h2')).toContainText('Select a file to upload');
+      const fileChooserPromise = page.waitForEvent('filechooser');
+      await page.getByLabel('Replace your previously').click();
+      const fileChooser = await fileChooserPromise;
+      await fileChooser.setFiles(
+        path.join(
+          __dirname,
+          '../../test-data/sample-sblar-files/errors-page-2-logic-few.csv',
+        ),
+      );
+      await expect(page.getByText('File upload in progress')).toBeVisible();
+      await expect(page.getByText('File upload successful')).toBeVisible({
+        timeout: 10_000,
+      });
+      await expect(page.getByText('Validation checks in progress')).toBeVisible(
+        {
+          timeout: 10_000,
+        },
+      );
+      await expect(
+        page.getByText('Errors were found in your file'),
+      ).toBeVisible({ timeout: 60_000 });
+
+      await page.getByRole('button', { name: 'Continue to next step' }).click();
```

Comment:
Seems like we're going to want an "upload file" action that we can reuse? I think everyone is modifying the code from the fixture, and it would be good to have one place to modify things, at least for these "happy path" uploads that aren't huge / expect errors?

Reply:
Good call. I created uploadFile.ts in the /utils folder.